core-bundle/src/Filesystem/FileDownloadHelper.php

```diff
@@ -115,14 +117,22 @@ public function handle(Request $request, VirtualFilesystemInterface $storage, \C
             }
         }
 
-        $response = new StreamedResponse(
-            static function () use ($storage, $file): void {
-                stream_copy_to_stream(
-                    $storage->readStream($file->getPath()),
-                    fopen('php://output', 'w')
-                );
-            }
-        );
+        $stream = $storage->readStream($file->getPath());
+        $metadata = stream_get_meta_data($stream);
+
+        // Prefer sendfile for local resources
+        if ('STDIO' === $metadata['stream_type'] && 'plainfile' === $metadata['wrapper_type'] && Path::isAbsolute($localPath = $metadata['uri'])) {
```

Comment:
Should we maybe extract that in a static util class? Or on the VFS maybe? Afaik it's duplicated here and in the `FigureBuilder`? /cc @ausi

Reply:
IMHO we are going to drop it in the `FigureBuilder` as soon as `contao/image` is capable of streaming.